components/camera/ultrasonic/ultrasonic.go

```diff
@@ -0,0 +1,99 @@
+package ultrasonic
+
+import (
+	"context"
+	"errors"
+	"image"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/components/sensor"
+	ultrasense "go.viam.com/rdk/components/sensor/ultrasonic"
+	pointCloud "go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/resource"
+
+	"github.com/edaniels/golog"
+	"github.com/viamrobotics/gostream"
+)
+
+var model = resource.DefaultModelFamily.WithModel("ultrasonic")
+
+type ultrasonicWrapper struct {
+	usSensor sensor.Sensor
+}
+
+func init() {
+	resource.RegisterComponent(
+		camera.API,
+		model,
+		resource.Registration[camera.Camera, *ultrasense.Config]{
+			Constructor: func(
+				ctx context.Context,
+				deps resource.Dependencies,
+				conf resource.Config,
+				logger golog.Logger,
+			) (camera.Camera, error) {
+				newConf, err := resource.NativeConfig[*ultrasense.Config](conf)
+				if err != nil {
+					return nil, err
+				}
+				return newCamera(ctx, deps, conf.ResourceName(), newConf, logger)
+			},
+		})
+}
+
+func newCamera(ctx context.Context, deps resource.Dependencies, name resource.Name, newConf *ultrasense.Config, logger golog.Logger) (camera.Camera, error) {
+
+	// sns, err := sensor.FromDependencies(deps, conf.Name)
```

Comment:
This is my attempt to use FromDependencies, but obviously it's not in use for right now.

Reply:
The way we are implementing this model is that you'll choose to build your ultrasonic sensor as either a sensor XOR a camera, so you don't have to worry about depending on an already built ultrasonic sensor. You can delete this part